Packs/SOCRadar/Integrations/FeedSOCRadarThreatFeed/FeedSOCRadarThreatFeed.py

```diff
@@ -0,0 +1,432 @@
+                    "type": indicator_type,
+                    "value": indicator,
+                    "rawJSON": {
+                        "value": indicator,
+                        "type": indicator_type
+                    },
+                    "fields": {
+                        "firstseenbysource": date_string_to_iso_format_parsing(first_seen_date),
+                        "lastseenbysource": date_string_to_iso_format_parsing(last_seen_date),
+                        "collection_maintainer_name": maintainer_name,
+                        "extra_info": extra_info
+                    }
+                }
+                if self.tags:
+                    indicator_obj["fields"]["tags"] = self.tags
+
+                if self.tlp_color:
+                    indicator_obj["fields"]["trafficlightprotocol"] = self.tlp_color
+
+                parsed_indicators.append(indicator_obj)
+
+        return parsed_indicators
+
+    def build_iterator(self, collection_name, limit=None, is_check_last_fetch=True) -> List:
+        """Builds a list of indicators.
+
+        :type collection_name: ``str``
+        :param collection_name: The name of the collection to fetch indicators from SOCRadar.
+
+        :type limit: ``int``
+        :param limit: Maximum number of indicators to fetch.
+
+        :type is_check_last_fetch: ``bool``
+        :param is_check_last_fetch: Flag to decide whether the last fetch should be checked or not.
+
+        :return: A list of JSON objects representing indicators fetched from a feed.
+        :rtype: ``list``
+        """
+        parsed_indicators = []
+        offset = 0
+        error_count = 0
+        batch_size = MAX_INDICATOR_FETCH_NUMBER
+        last_fetch_dict = demisto.getIntegrationContext().get('last_fetch', {})
+        while True:
+            if limit is not None:
+                if offset >= limit:
+                    break
+                batch_size = min(limit - offset, MAX_INDICATOR_FETCH_NUMBER)
+            try:
+                raw_response = self.get_collection_indicators(collection_name, offset, batch_size)
+                if raw_response.get('is_success'):
+                    collection_dict = raw_response.get('data', {}).get(collection_name, {})
+                    raw_indicators_list = collection_dict.get('collection_data_list', [])
+                    collection_date = collection_dict.get('collection_date')
+                    collection_feed_type = collection_dict.get('collection_feed_type')
+                    if not raw_indicators_list:
+                        break
+                    if is_check_last_fetch:
+                        if last_fetch := last_fetch_dict.get(collection_name):
+                            collection_date = datetime.strptime(collection_date, '%Y-%m-%d').date()
+                            last_fetch = datetime.strptime(last_fetch, '%Y-%m-%d').date()
+                            if last_fetch >= collection_date:
```

Comment:
Yes, the indicators from SOCRadar service are provided in reverse chronological order so the logic in this piece of code should work fine. Otherwise the while loop might have been finished earlier depending on the received unsorted order of the collection date. But we don't have this problem here.